src/query/plan/cost_estimator.hpp

```diff
@@ -62,6 +62,8 @@ class CostEstimator : public HierarchicalLogicalOperatorVisitor {
     static constexpr double kEdgeUniquenessFilter{1.5};
     static constexpr double kUnwind{1.3};
     static constexpr double kForeach{1.0};
+    static constexpr double kUnion{1.0};
+    static constexpr double kSubquery{1.0};
```

Comment:
Is somehow taken into account that cost of having subqueries can have large cost since it is embedded inside the outer query?

Reply:
yes, the cost of subquery is 1 since it will execute once, or depending on the outer cardinality. it does not add any additional complexity or cardinality, but it propagates it from the inner logic